microshift.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+source tools.sh
+source snc-library.sh
+
+BUNDLE_TYPE="microshift"
+INSTALL_DIR=crc-tmp-install-data
+CRC_VM_NAME=${CRC_VM_NAME:-crc}
+BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp}
+OPENSHIFT_RELEASE_VERSION=4.12.0
+
+if grep -q -i "release 9" /etc/redhat-release
+then
+  echo "Check if system is registered"
+  # Check the subscription status and register if necessary
+  if ! sudo subscription-manager status >& /dev/null ; then
+     sudo subscription-manager register
+  fi
+else
+  echo "This script only works for RHEL-9"
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+sudo virsh destroy ${CRC_VM_NAME} || true
+sudo virsh undefine --nvram ${CRC_VM_NAME} || true
+sudo rm -fr /var/lib/libvirt/images/${CRC_VM_NAME}.qcow2
+sudo rm -fr /var/lib/libvirt/images/microshift-installer-*.iso
+
+# Download the oc binary for specific OS environment
+OC=./openshift-clients/linux/oc
+download_oc
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+# Configure host (RHEL-9) with required internal repo
+sudo tee /etc/yum.repos.d/latest-fdp.repo >/dev/null <<EOF
+[latest-FDP]
+name = latest-FDP-Buildroot-rpms
+baseurl = http://download.eng.bos.redhat.com/rhel-9/nightly/FDP/latest-FDP-9-RHEL-9/compose/Server/\$basearch/os/
+enabled = 1
+gpgcheck = 0
+EOF
+sudo tee /etc/yum.repos.d/latest-rhaos.repo >/dev/null <<EOF
+[latest-RHAOS]
+name = latest-RHAOS-puddle
+baseurl = http://download.lab.bos.redhat.com/rcm-guest/puddles/RHAOS/plashets/4.12-el9/building/\$basearch/os/
+enabled = 1
+gpgcheck = 0
+EOF
+
+rm -fr microshift
+# Clone microshift repo
+git clone -b release-4.12 https://github.com/openshift/microshift.git
```

Comment:
With microshift being officially released, I guess this can now be updated to follow https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/installing/microshift-embed-in-rpm-ostree or https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/installing/microshift-install-rpm more closely?

Reply:
@cfergeau updated as per installation doc, now we are consuming released rpm instead building one. we still need to do cloning of microshift repo to consume the image-builder scripts and kickstart files.